server/src/main/java/org/elasticsearch/transport/LeakTracker.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.transport;
```

Comment:
Does this need to be in `server/src/main` or could we have it in `test/framework` instead? Not sure if you have plans to use it more generally in future, but if we're not going to test it we really should keep it out of production code.

Reply:
Not really, at least not right now. I used this a lot via ad-hoc testing for all kinds of classes when working on https://github.com/elastic/elasticsearch/pull/67502 that's how it ended up in prod-code, but for now it's only used in tests -> moved it :)